src/lib/tabs/tab-group.ts

```diff
@@ -111,7 +111,7 @@ export class MatTabGroup extends _MatTabGroupMixinBase implements AfterContentIn
 
   /** The index of the active tab. */
   @Input()
-  set selectedIndex(value: number | null) { this._indexToSelect = value; }
+  set selectedIndex(value: number | null) { this._indexToSelect = coerceNumberProperty(value); }
```

Comment:
it looks like the author was intentionally trying to allow `null` here in which case I think you want `coerceNumberProperty(value, null)`

Reply:
The `coerceNumberProperty` signature does not support `null` as the second parameter. Should we extend it you think to support that?

Reply:
Hmm, in a lot of cases people really do just want number. What if we did this?

```ts
export function coerceNumberProperty(value: any): number;
export function coerceNumberProperty<D>(value: any, fallback: D): number | D;
export function coerceNumberProperty(value: any, fallbackValue = 0) { ... }
```

It would allow people to get just a `number` in 1-arg case and `number|whatever` in 2-arg case. @jelbourn WDYT?

Reply:
That seems good to me

Reply:
Works well.